tfhe/docs/tutorials/ascii_fhe_string.md

```diff
@@ -0,0 +1,156 @@
+# A first complete example: FheLatinString (Integer)
+
+The goal of this tutorial is to build a data type that represents a ASCII string in FHE while implementing the `to_lower` and `to_upper` functions.
+
+An ASCII character is stored in 7 bits.
+To store an encrypted ASCII we use the `FheUint8`.
+
+* The uppercase letters are in the range \[65, 90]
+* The lowercase letters are in the range \[97, 122]
+
+`lower_case = upper_case + UP_LOW_DISTANCE` <=> `upper_case = lower_case - UP_LOW_DISTANCE`
+
+Where `UP_LOW_DISTANCE = 32`
+
+
+## Types and methods.
+
+This type will hold the encrypted characters as a `Vec<FheUint8>` to implement the functions that change the case.
+
+To use the `FheUint8` type, the `integer` feature must be activated:
+
+```toml
+# Cargo.toml
+
+[dependencies]
+# Default configuration for x86 Unix machines:
+tfhe = { version = "0.4.0", features = ["integer", "x86_64-unix"]}
+```
+
+Other configurations can be found [here](../getting_started/installation.md).
+
+
+
+In the `FheLatinString::encrypt` function, some data validation is done:
+
+* The input string can only contain ascii characters.
+
+It is not possible to create branches where the condition if an encrypted integer, meaning the function cannot use conditional statements.
+Checking if the 'char' is an uppercase letter to modify it to a lowercase one cannot be done using a branch, like this:
+
+```rust
+pub const UP_LOW_DISTANCE: u8 = 32;
+
+fn to_lower(c: u8) -> u8 {
+    if c > 64 && c < 91 {
+        c + UP_LOW_DISTANCE
+    } else {
+        c
+    }
+}
+```
+
+We can remove the branch this way:
+
+```rust
+pub const UP_LOW_DISTANCE: u8 = 32;
+
+fn to_lower(c: u8) -> u8 {
+    c + (c > 64) as u8 * (c < 91) as u8 * UP_LOW_DISTANCE
+}
+```
+
+On an homomorphic integer, this gives
+
+```rust
+use tfhe::prelude::*;
+use tfhe::FheUint8;
+
+pub const UP_LOW_DISTANCE: u8 = 32;
+
+fn to_lower(c: &FheUint8) -> FheUint8 {
+    c + (c.gt(64)) * (c.lt(91)) * UP_LOW_DISTANCE
+}
+```
+
+The whole code is:
+
+```rust
+use tfhe::prelude::*;
+use tfhe::{generate_keys, set_server_key, ClientKey, ConfigBuilder, FheUint8};
+
+pub const UP_LOW_DISTANCE: u8 = 32;
+
+struct FheAsciiString {
+    bytes: Vec<FheUint8>,
+}
+
+fn to_upper(c: &FheUint8) -> FheUint8 {
+    c - (c.gt(96)) * (c.lt(123)) * UP_LOW_DISTANCE
+}
+
+fn to_lower(c: &FheUint8) -> FheUint8 {
+    c + (c.gt(64)) * (c.lt(91)) * UP_LOW_DISTANCE
```

Comment:
I think here using `&` instead of `*` in `(c.gt(96)) * (c.lt(123))` might be faster, but it would need to be tried

Reply:
Done
I couldn't see a timings difference

Reply:
ok